lib/chef/knife/vsphere_vm_snapshot.rb

```diff
@@ -41,23 +45,50 @@ class Chef::Knife::VsphereVmSnapshot < Chef::Knife::BaseVsphereCommand
          :description => "Indicates whether to start the VM after a successful revert",
          :boolean => false
 
-  def run
+  option :memory,
+	 :long => "--memory",
+	 :description => "Indicates whether to dump the internal memory state in the snapshot",
+	 :boolean => false
+
+  def find_vm_in_folder(folder, vmname)
+    path = (folder.path[3..-1].map { |x| x[1] }.* '/')
+    children = folder.children.find_all
+    children.each do |child|
+      if child.class == RbVmomi::VIM::VirtualMachine && child.name == vmname
+	puts "Found #{child.name} in #{folder.name}"
+        return folder
+      elsif child.class == RbVmomi::VIM::Folder
+        vm = find_vm_in_folder(child, vmname)
+        if vm then return vm end
+      end
+    end
+    return false
+  end
+
 
+  def run
     $stdout.sync = true
 
+    vim = get_vim_connection
+
     vmname = @name_args[0]
+
     if vmname.nil?
       show_usage
       ui.fatal("You must specify a virtual machine name")
-      exit 1
+      abort
     end
 
-    vim = get_vim_connection
-
-    baseFolder = find_folder(get_config(:folder));
+    if (get_config(:find))
+      puts "No folder enterered, searching for #{vmname}"
+      srcFolder = find_folder(get_config(:folder));
```

Comment:
I'm not sure if Rubocop will mention this, but for the most part we're using `snake_case` instead of `camelCase` for variable names.